File: task-backend/app/models/task.rb
class Task
  include Mongoid::Document
  include Mongoid::Timestamps

  field :title, type: String
  field :description, type: String
  field :status, type: String, default: "pending"

  validates :title, presence: true, uniqueness: true
  validates :status, inclusion: { in: %w[pending completed] }
end

File: task-frontend/src/app/models/task.ts
export interface Task {
  id: string;
  title: string;
  description: string | null;
  status: 'pending' | 'completed';
  created_at: string;
  updated_at: string;
}

export interface TasksResponse {
  message: string;
  data: Task[];
}

export interface TaskResponse {
  message: string;
  data: Task;
}

export interface TaskFormPayload {
  title: string;
  description?: string;
  status?: 'pending' | 'completed';
}